libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/resource/ResourceLoader.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.navigation.ui.utils.resource
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.common.ResourceLoadError
+import com.mapbox.common.ResourceLoadProgress
+import com.mapbox.common.ResourceLoadResult
+
+/**
+ * ResourceLoader is responsible for downloading and storage of any Nav SDK BLOB assets.
+ */
+abstract class ResourceLoader {
```

Comment:
Why `class` instead of `interface` when all functions are abstract?

Reply:
By making this an `abstract class` we will be able to modify the interface without breaking semver.